com.microsoft.mrtk.input/Utilities/EyeCalibrationChecker.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices.WindowsRuntime;
+using UnityEngine;
+using UnityEngine.Events;
+#if WINDOWS_UWP
+using Windows.Perception;
+using Windows.Perception.People;
+using Windows.Perception.Spatial;
+using Windows.UI.Input.Spatial;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// A helper class used to check eye calibration status. Only for UWP Platforms. 
+    /// </summary>
+    [AddComponentMenu("MRTK/Input/Eye Calibration Checker")]
+    public class EyeCalibrationChecker : MonoBehaviour
+    {
+        #region Public Enums
+
+        public enum EyeCalibrationStatus
+        {
+            /// <summary>
+            /// The eye calibration status is not defined. 
+            /// </summary>
+            Undefined,
+
+            /// <summary>
+            /// The eye calibration status could not be retrieved because this is a non-UWP device.
+            /// </summary>
+            IsNonUWPDevice,
+
+            /// <summary>
+            /// The eye calibration status could not be retrieved becaause eyes are not being tracked.
+            /// This occurs when SpatialPointerPose's Eyes property is null. 
+            /// </summary>
+            IsNotTracked,
+
+            /// <summary>
+            /// The eye calibration check has expired. Assume that eyes are not calibrated.
+            /// </summary>
+            IsExpired,
+
+            /// <summary>
+            /// The eye calibration status was retrieved and eyes are not calibrated. 
+            /// </summary> 
+            IsNotCalibrated,
+
+            /// <summary>
+            /// The eye calibration status was retrieved and eyes are calibrated. 
+            /// </summary>
+            IsCalibrated
+        };
+
+        #endregion Public Enums
+
+        #region Serialized Fields
+
+        /// <summary>
+        /// For testing purposes, you can manually assign whether eyes are calibrated or not in editor. 
+        /// </summary>
+        [field: SerializeField, Tooltip("For testing purposes, you can manually assign whether eyes are calibrated or not in editor.")]
+        public EyeCalibrationStatus EditorTestIsCalibrated = EyeCalibrationStatus.IsCalibrated;
+
+        #endregion Serialized Fields
+
+        #region Private Fields
+        private EyeCalibrationStatus calibrationStatus;
+
+        /// <summary>
+        /// Tracks whether eyes are present and calibrated. 
+        /// </summary>
+        public EyeCalibrationStatus CalibratedStatus
+        {
+            get => calibrationStatus;
+        }
+
+        private EyeCalibrationStatus prevCalibrationStatus;
+        #endregion Private Fields
+
+        #region Events
+        /// <summary>
+        /// Event fired when IsCalibrated changes from false to true.
+        /// </summary>
+        public UnityEvent OnEyeCalibrationDetected = new UnityEvent();
```

Comment:
I think having a 3rd event would be useful too. 

```
[SerializedField]
[Tooltip("...."]
private EyeTrackingStatusEvent calibratedStatusChanged = new EyeTrackingStatusEvent();

public EyeTrackingStatusEvent CalibratedStatusChanged => calibratedStatusChanged ;
```
Which would require these types

```
[Serializable]
struct EyeTrackingStatusEventArgs
{
   public EyeCalibrationStatus CalibratedStatus { get; set; }
}

[Serializable]
class EyeTrackingStatusEvent : UnityEvent<EyeTrackingStatusEventArgs>
{
}
```